test/test_encryption.py

```diff
@@ -1358,5 +1360,230 @@ def test_automatic(self):
             expected_document_extjson, {"secret_gcp": "string0"})
 
 
+class TestDeadlock(EncryptionIntegrationTest):
```

Comment:
Is this a prose test that's documented in the spec? If so can you link to it in a comment?

Reply:
Done.